controllers/storageclient_controller.go

```diff
@@ -206,6 +219,43 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClientReconciler) reconcileProviderConfig(storageClientResponse *providerpb.StorageConfigResponse) error {
+
+	for _, eResource := range storageClientResponse.ExternalResource {
+		// Create the received resources, if necessary.
+		switch eResource.Kind {
+		case "ClusterResourceQuota":
+			if err := r.reconcileClusterResourceQuota(eResource); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *StorageClientReconciler) reconcileClusterResourceQuota(extResource *providerpb.ExternalResource) error {
```

Comment:
We should limit the places in code where we depend on proto buf structs. We don't want to make the entirety of the flow coupled with our network protocol. 

Move the unmarshalling into the call site and send in only the desired state.